src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -76,12 +80,25 @@ class BasePaymentsPage extends React.Component {
             this.setMenuPosition();
         }
 
+        // If the user was previously offline, skip debouncing showing the loader
+        if (prevProps.network.isOffline && !this.props.network.isOffline) {
+            this.setShouldShowLoadingSpinner();
+        } else {
+            this.debounceSetShouldShowLoadingSpinner();
+        }
+
+        // previously online OR currently offline, skip fetch
         if (!prevProps.network.isOffline || this.props.network.isOffline) {
             return;
         }
+
         this.fetchData();
     }
 
+    setShouldShowLoadingSpinner() {
+        this.setState({shouldShowLoadingSpinner: this.props.isLoadingPaymentMethods && !this.props.network.isOffline});
```

Comment:
Calling setState causes the component to update. Whenever the component updates we call this function and setState, so we have a loop.

Reply:
I think we can fix this by only setting `shouldShowLoadingSpinner` if its value changes.